lib/api/v1/assignment.rb

```diff
@@ -391,6 +401,19 @@ def assignment_json(assignment, user, session, opts = {})
     hash
   end
 
+  def can_show_statistics?(assignment, submission, user)
+    # To show statistics, the student's grade must be visible to the student
+    return false unless (submission.score.present? && !submission.hide_grade_from_student?)
+
+    # and the assignment must have points_possible > 0,
+    return false unless (assignment.points_possible.present? && assignment.points_possible > 0)
+
+    # Students can only see statistics when count >= 5 and not disabled by the instructor
+    # Instructor can see statistics at any time.
+    count = assignment.score_statistic&.count || 0
+    assignment.context.grants_right?(user, :read_as_admin) || (count >= 5 && !assignment.context.hide_distribution_graphs)
```

Comment:
It looks like the "count >= 5" logic here is based on the logic in `GradeSummaryAssignmentPresenter#hide_distribution_graphs?` , is that right? It'd be great if we could find a way to abstract the logic into a shared place, so that we can DRY things up a bit (also, that way, if we decide down the road that we want to change the threshold from 5 to something else, we have one place to change it).

Reply:
Yeah that's where I got the logic from. I moved the logic to the assignment and submission model and it's shared now.